term/src/data.rs

```diff
@@ -0,0 +1,56 @@
+//! Types that can be used generally to hold the data of RDF terms, e.g. the
+//! lexical value of a literal or an IRI.
+
+mod _impl;
+
+use std::hash::Hash;
+
+/// Trait alias for types holding the textual data of terms.
+///
+/// The `TermData` considered in this crate are: `&str`, `String`, `Box<str>`,
+/// `Rc<str>`, `Arc<str>`, Cow<\'_, str> and `MownStr<\'_>` and the respective
```

Comment:
Do you really have use for `&Box<str>` or `&Arc<str>` used as `TermData`?

Since all `TermData` types implement `AsRef<str>`, it seems to me that `&str` is sufficient whenever you simply need to borrow the underlying string...